core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/CompressRequestStage.java

```diff
@@ -177,9 +189,8 @@ private static boolean resolveRequestCompressionEnabled(RequestExecutionContext
     private static boolean isRequestSizeWithinThreshold(SdkHttpFullRequest.Builder input, RequestExecutionContext context) {
         int minimumCompressionThreshold = resolveMinCompressionSize(context);
         validateMinCompressionSizeInput(minimumCompressionThreshold);
-
-        long contentLength = Long.parseLong(input.firstMatchingHeader("Content-Length").orElse("0"));
-        return contentLength >= minimumCompressionThreshold;
+        int requestSize = SdkBytes.fromInputStream(input.contentStreamProvider().newStream()).asByteArray().length;
```

Comment:
We are reading the entire stream here  to get the request size,
We should  avoid intermediate reading  of the streams this can cause performance impact , can we discuss this with team ?

Reply:
the content length header isn't set yet for nonstreaming operations. not sure how we would get the content length if not by reading the stream, can discuss with team

Reply:
updated method to check for content length header first, and if not present, then read the stream to determine length